src/markdownbox.jl

```diff
@@ -158,6 +158,13 @@ function render_element(md::Markdown.HorizontalRule, l::MarkdownBox, idx)
 end
 
 
+function render_element(md::Markdown.LaTeX, l::MarkdownBox, idx)
+    latex = latexstring(strip(md.formula))
```

Comment:
This is great - MakieTeX will work seamlessly here!

Sidenote - what should I do if I want to inject some tikz into a presentation?  Specifically, I would like to inject the tikzpicture environment into Markdown somehow, but am not sure how I'd do so.

Reply:
AFAIK `Makie` uses `MathTeXEngine` for rendering `LaTeX` strings, or?
So for `MakieTeX` to work we have to add extra support, but I think that shouldn't be difficult. The question is how to decide whether we want to render with one or the other.

Regarding tikz: Haven't thought about that yet. If it is as easy as just calling a label with a valid `LaTeXString` then perhaps the following might work
```julia
fig = Figure()
MarkdownBox(fig[1,1], md"""
```math
\begin{tikzpicture}
   ...
\end{tikzpicture}
\\``` # don't know how to escape ``` inside a ```...``` block
""")
```

Alternatively, if you know how to render tikz with Makie alone, then you could also just leave a gap between your MarkdownBoxes, e.g.
```julia
MarkdownBox(fig[1,1], md"""
# Crazy things need to be sketched
""")
# render tikz pic into fig[2,1]
MarkdownBox(fig[3,1], md"""
Can you believe this?
""")
```

Reply:
Well I can call a label with a LaTeXString, but it can't have the `$$` math delimiters around it.  I'm not sure how you parse the ```math``` environment but I imagine it triggers display math mode in latex.

I think your second solution would be best!  I can pass a MakieTeX object directly to Label which will render well.

In v0.2, I refactored MakieTeX to pirate the plotting methods for text and override what MathTeXEngine is used for in Makie.  Basically, any LaTeXString rendering in `text` is forwarded to a MakieTeX internal recipe instead of using MathTeXEngine.

Reply:
I think you can use the `LaTeXString` constructor to avoid placing your code inside `$$`, see last paragraph at https://github.com/stevengj/LaTeXStrings.jl.
The `math` block in `Markdown` gets parsed into a `Markdown.LaTeX` element which then only stores the content within that block, e.g. without `$$`. So what is `md.formula` above would just be whatever you put into the ```math ... ``` block, but without the `$$`.

I have to say that the above suggestions were the first things that came to my head. I am open for alternative solutions, since this is clearly a hack. Perhaps #4 could help here because (I think) it allows you to add metadata to Markdown elements directly.

> In v0.2, I refactored MakieTeX to pirate the plotting methods for text and override what MathTeXEngine is used for in Makie.

So if I do `using MakieTeX` before `using MakieSlides`, then things Latex elements will be rendered with a true Latex engine? That's amazing :)

Reply:
> So if I do using MakieTeX before using MakieSlides, then things Latex elements will be rendered with a true Latex engine?

Yup!  The contents only re-render when the actual TeX changes (or the density of the rendered image for GLMakie), so it's pretty convenient.  It's integrated directly with Label (and has its own scalable layoutable `LTeX`), and responds to theming if you pass a `LaTeXString(raw_tex)`.  I haven't put font support in yet, but that would basically be a lookup table from font names to their respective LaTeX packages.